### ISSUE TITLE
Custom output layer / loss function

### ISSUE THREAD

**Issue by vcoscrato** [keras-team/keras#11579]
I am new to the keras world and i am trying to build a feedfoward net which output should be a vector of parameters to be used in a linear ensembler, i want to minimize the MSE for the ensembler, then, i created the following loss function,

`def loss(theta, y_train):
    return np.sum(np.square(np.dot(g_data, theta) - y_train))`

g_data is my matrix for single estimators predictions, which multiplyed to theta (net output) gives me the real predictions. Next, i create the model as follows,

`model2 = Sequential()
model2.add(Dense(200, activation='relu'))
model2.add(Dense(len(estimators), activation='linear'))
model2.compile(loss=loss, optimizer='adam')
model2.fit(x_train, y_train, epochs=2, batch_size=32, verbose=1)`

Of course it does not run, cause y_train is a single output meanwhile i want to train a vector of outputs, as well, i would have to pass the sample index for each batch in order to use the proper rows of g_data in the loss function calculation.

Any ideas on how to working around this?
